LCM/dsc/engine/ConfigurationManager/LocalConfigManagerHelper.c

```diff
@@ -2075,31 +2191,51 @@ MI_Result ApplyPendingConfig(
         result = ClearBuiltinProvCache(PACKAGE_PROVIDER_CACHE, cimErrorDetails);
         if (result == MI_RESULT_OK)
         {
+            flags |= LCM_EXECUTE_APPLYNEWCONFIG;
             result = ApplyConfig(lcmContext, GetPendingConfigFileName(), moduleManager, flags, resultStatus, cimErrorDetails);
         }
     }
 
-
     if (result != MI_RESULT_OK)
     {
-	// Attempt to save a Failed configuration file
-	CopyConfigurationFile(CONFIGURATION_LOCATION_PENDING, CONFIGURATION_LOCATION_FAILED, MI_TRUE, cimErrorDetails);
+	    // Attempt to save a Failed configuration file
+	    CopyConfigurationFile(CONFIGURATION_LOCATION_PENDING, CONFIGURATION_LOCATION_FAILED, MI_TRUE, cimErrorDetails);
         RetryDeleteFile(GetPendingConfigFileName());
         File_RemoveT(GetConfigChecksumFileName());
         return result;
     }
 
-    if (DSC_RESTART_SYSTEM_FLAG & *resultStatus)
+    //If restart requested or test flag is set, do not move to current.mof, and do not delete the pending mof file.
+    if (DSC_RESTART_SYSTEM_FLAG & *resultStatus || flags & LCM_EXECUTE_TESTONLY)
     {
         return result;
     }
+
     //If whatif flag is set, do not move to current.mof, just delete the pending mof file.
-    if(lcmContext->executionMode & LCM_SETFLAGS_ENABLEWHATIF)
+    if (lcmContext->executionMode & LCM_SETFLAGS_ENABLEWHATIF)
     {
         RetryDeleteFile(GetPendingConfigFileName());
         return result;
     }
-    return MoveConfigurationFiles(cimErrorDetails);
+
+	result = GetMetaConfig(&metaConfigInstance);
+	EH_CheckResult(result);
+
+	result = DSC_MI_Instance_GetElement(metaConfigInstance, MSFT_DSCMetaConfiguration_ConfigurationMode, &configModeValue, NULL, NULL, NULL);
+	EH_CheckResult(result);	
```

Comment:
1) Tabs
2) Can we move this code before applying config as if this fails we don't need to do hard work.